src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CSharp/TypeAccessibilityCodeActionProvider.cs

```diff
@@ -143,7 +162,7 @@ internal class TypeAccessibilityCodeActionProvider : CSharpCodeActionProvider
             var fqnCodeAction = CreateFQNCodeAction(context, diagnostic, codeAction, fqn);
             typeAccessibilityCodeActions.Add(fqnCodeAction);
 
-            var addUsingCodeAction = CreateAddUsingCodeAction(context, fqn);
+            var addUsingCodeAction = AddUsingsCodeActionProviderFactory.CreateAddUsingCodeAction(fqn, context.Request.TextDocument.Uri);
```

Comment:
Yes, as is for [VSCode](https://github.com/dotnet/aspnetcore-tooling/pull/2684/files/184da239ba42d1525f5d7c2438cc64d788fd922e#diff-0be4ed27bb2b83f2a3854b8612e1c9b8d1779a500984a97102d074a6afec2f70R127). 

I don't believe the roslyn changes will propagate on the VSCode side as O# generates the code actions in a different way. The add using isn't an issue on the VSCode side as instead of providing `using Data` like VS/roslyn did, VSCode/O# provides `using blazorserver.Data` (ie. fully qualified).

Reply:
> The add using isn't an issue on the VSCode side as instead of providing `using Data` like VS/roslyn did, VSCode/O# provides `using blazorserver.Data` (ie. fully qualified).

Okay... so looks like VSCode made some changes as this is now an issue for VSCode as well. We now get a codeaction for `using Data` instead of `using blazorserver.Data`. This will be solved by moving O#/VSCode to using the `textDocument/codeAction` endpoints discussed earlier.